lib/linalg/Matrix.cpp

```diff
@@ -1788,25 +1788,31 @@ const
 }
 
 void
-Matrix::orthogonalize(double zero_tol)
+Matrix::orthogonalize(bool double_pass, double zero_tol)
 {
     for (int work = 0; work < d_num_cols; ++work)
     {
-        // Orthogonalize the column.
-        for (int col = 0; col < work; ++col)
+        // Orthogonalize the column (twice if double_pass == true).
+        for (int k = 0; k < 2; k++)
```

Comment:
This is probably a bit picky, but to try to reduce the number of if-statements inside the outer `work` loop, we could define the number of passes at the top of the function outside of the loop:
`const int num_pass = double_pass ? 2 : 1;`
Then the `k` loop could then be re-written as `for (int k = 0; k < num_pass; k++)` and the `if (!double_pass) break;` line can be removed.

Reply:
Done.